workorders.lic

```diff
@@ -332,51 +332,80 @@ class WorkOrders
     end
   end
 
-  def count_combine_rem(stock_room, herb, herb_stock, quantity, herb2)
-    if inside?("first #{herb}", @bag) && inside?("second #{herb}", @bag)
-      bput("get #{herb} from my #{@bag}", 'You get')
-      bput("get #{herb} from my #{@bag}", 'You get')
-      case bput('combine', 'You combine', 'That stack of herbs')
-      when 'That stack of herbs'
-        stow_tool(left_hand)
-        stow_tool(right_hand)
-      when 'You combine'
-        case bput("get #{herb} from my #{@bag}", 'You get', 'What were')
-        when 'You get'
-          case bput('combine', 'You combine', 'That stack of herbs')
-          when 'That stack of herbs', 'You combine'
-            stow_tool(left_hand)
-            stow_tool(right_hand)
-          end
+  # Function which will attempt to combine any stacks of herbs in your inventory, will count them, and determine if more need to be ordered.
+  def count_combine_rem(stock_room, quantity, herb, herb_stock)
+  # Initialize variables
+    found_stack = true;
+    herb_volume_total = 0;
+    last_herb_volume = 0;
+    last_descriptor = "";
+    stack_descriptor = "";
+
+    # We have to only ever use the last word in a multi-word herb.  We will need to be careful when counting because of this.
+    # This is to resolve "red flower" vs "blue flower" because "count third red flower in my bag" fails.
+    # We must instead do a generic "tap third flower in my bag" and determine if we're tapping the right kind of flowers from the response, then actually count them.
+    herb_for_tapping = (herb.gsub(/\s+/m, ' ').strip.split(" ")).last
```

Comment:
I don't believe we should break out the `adjective` and `noun` in the recipe YAML for 2 reasons.
1. Because it doesn't help solve the problem that I outline in the notes here.  We would still need to perform the tapping before the counting because of how DR fails to recognize the name of the herb in the first place when it's inside a container.  There would be nothing gained in terms of reducing logic.
2. The only place we need the `adjective` and the `noun` split is inside the `count_combine_rem` function.  Looking in the `remedy_items` function, which serves as the core logic here, we can see that we actually need the combined herb name in several places when we call in to the `;remedy` script.  This means that in most places, the whole herb name is the useful keyword, and only in one small place do we need the herb split apart.  Additionally, I would argue since so few of the herbs actually require this logic to function, it seems unnecessary to make such a low level change just to accommodate.